### PR TITLE
fix(aio): upgrade - remove broken hybrid router example 

### PR DESCRIPTION
**Pull request by filipesilva**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Other... Please describe: AIO ngUpgrade fix
```

**What is the current behavior?** (You can also link to an open issue here)
The current ngUpgrade guide setup has issues with zones for dual router apps.

**What is the new behavior?**
Dual router advice in ngUpgrade guide is removed, and bootstrap instructions are changed.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
This is a ported PR from the angular/angular.io repo https://github.com/angular/angular.io/pull/3541, augmented with example removal.

/cc @wardbell @petebacondarwin @Foxandxss

### CODE REVIEW

**Comment by wardbell** [comment]
@petebacondarwin should review the code changes (and some prose) to confirm that this is what we should be saying.

I noticed a few small prose things I'd like to fix. The most important, though, is that it hasn't gone through the "we" -> "you" sweep. I'll take care of that in an attached commit.

File: aio/content/examples/upgrade-module/src/app/divide-routes/app.module.ts [hunk 1]
```diff
@@ -31,10 +31,17 @@ class HybridUrlHandlingStrategy implements UrlHandlingStrategy {
     // use custom url handling strategy
     { provide: UrlHandlingStrategy, useClass: HybridUrlHandlingStrategy }
   ],
-  declarations: [ AppComponent ],
-  bootstrap: [ AppComponent ]
+  declarations: [ AppComponent ]
 })
-export class AppModule { }
+export class AppModule extends UpgradeModule {
```
Comment (petebacondarwin) [inline aio/content/examples/upgrade-module/src/app/divide-routes/app.module.ts]: No need for the extends here.

File: aio/content/examples/upgrade-module/src/app/divide-routes/app.module.ts [hunk 2]
```diff
@@ -31,10 +31,17 @@ class HybridUrlHandlingStrategy implements UrlHandlingStrategy {
...
+  constructor(injector: Injector, zone: NgZone) {
+    super(injector, zone);
```
Comment (petebacondarwin) [inline aio/content/examples/upgrade-module/src/app/divide-routes/app.module.ts]: no need for the super call here (once we remove the extends above.

File: aio/content/examples/upgrade-module/src/app/upgrade-static/hero-detail.component.ts [hunk 3]
```diff
@@ -21,5 +22,15 @@ export class HeroDetailDirective extends UpgradeComponent {
...
     super('heroDetail', elementRef, injector);
   }
+
+  // #enddocregion hero-detail-upgrade
+  // For this class to work when compiled with AoT, we must implement these lifecycle hooks
+  // because the AoT compiler will not realise that the super class implements them
+  ngOnInit() { super.ngOnInit(); }
+  ngOnChanges(changes: SimpleChanges) { super.ngOnChanges(changes); }
+  ngDoCheck() { super.ngDoCheck(); }
+  ngOnDestroy() { super.ngOnDestroy(); }
```
Comment (petebacondarwin) [inline aio/content/examples/upgrade-module/src/app/upgrade-static/hero-detail.component.ts]: I think that the latest Angular will actually cope without us manually providing these. @gkalpak  is that correct?

File: aio/content/guide/upgrade.md [hunk 4]
```diff
@@ -522,10 +521,16 @@ will result in the same thing:
 
 
 
-Now introduce Angular to the project. Inspired by instructions in
-[the Setup](guide/setup), you can selectively copy in material from the
+To begin converting our AngularJS application to a hybrid, we need to load the Angular framework.
+You can see how this can be done with SystemJS by following the instructions in [the Setup](setup.html),
```
Comment (petebacondarwin) [inline aio/content/guide/upgrade.md]: the link should be to `guide/setup`.

File: aio/content/guide/upgrade.md [hunk 5]
```diff
@@ -1158,83 +1161,16 @@ Module Factory:
...
-And that's all we need to get the full benefit of AoT for Angular apps!
-
-
-<div class="alert is-helpful">
-
-
-
-The AoT metadata collector will not detect lifecycle hook methods on a parent class' prototype,
-so in order for upgraded components to work we needs to implement the lifecycle hooks
+Finally, the AoT metadata collector does not detect lifecycle hook methods on the prototype of a parent class.
```
Comment (petebacondarwin) [inline aio/content/guide/upgrade.md]: We need to check whether this is still true. I am sure that @gkalpak told me it was no longer needed.

File: aio/content/guide/upgrade.md [hunk 6]
```diff
@@ -2244,47 +2173,33 @@ which recommends that you create a `NgModule` dedicated to router configuration
...
 That tells Angular that it should bootstrap the app with the _root_ `AppComponent` and
 insert it's view into the host web page.
 
-We can also remove the `ngDoBootstrap()` override from `app.module.ts` since we are now
-bootstrapping from Angular.
-
-And since `PhoneListComponent` isn't being rendered from a `<phone-list>` tag anymore,
-but rather routed to, we can do away with it's Angular selector as well.
+We must also remove the bootstrap of the AngularJS module from `ngDoBootstrap()` in `app.module.ts`
+and the `UpgradeModule` import.
```
Comment (petebacondarwin) [inline aio/content/guide/upgrade.md]: I am not sure that these two sentences are correct.

Comment (petebacondarwin) [inline aio/content/guide/upgrade.md]: Hmm, I see that what we are really describing now is a non-ngUpgrade application. But that doesn't really match the headings above.

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
Thanks for taking this on @filipesilva !

The bit where we have removed the dual-router still doesn't quite read correctly IMO.
There are a few other nitpicks and the PR needs rebasing and conflicts resolving.

File: aio/content/examples/upgrade-module/src/app/upgrade-static/hero-detail.component.ts [hunk 7]
```diff
@@ -21,5 +22,15 @@ export class HeroDetailDirective extends UpgradeComponent {
...
+  // #docregion hero-detail-upgrade
 }
-// #enddocregion hero-detail-upgrade
+// #enddocregion hero-detail-upgrade		  // #enddocregion hero-detail-upgrade
```
Comment (gkalpak) [inline aio/content/examples/upgrade-module/src/app/upgrade-static/hero-detail.component.ts]: Too much `enddocregion hero-detail-upgrade`?

File: aio/content/examples/upgrade-module/src/app/upgrade-static/hero-detail.component.ts (hunk 3 above)
Comment (gkalpak) [inline aio/content/examples/upgrade-module/src/app/upgrade-static/hero-detail.component.ts]: Correct.

**Comment by wardbell** [comment]
@petebacondarwin As I write, `yarn docs` hangs when run with this PR.

File: aio/content/examples/upgrade-module/src/app/divide-routes/app.module.ts (hunk 2 above)
Comment (filipesilva) [inline aio/content/examples/upgrade-module/src/app/divide-routes/app.module.ts]: Fixed

File: aio/content/examples/upgrade-module/src/app/divide-routes/app.module.ts (hunk 1 above)
Comment (filipesilva) [inline aio/content/examples/upgrade-module/src/app/divide-routes/app.module.ts]: Fixed

**Comment by filipesilva** [comment]
@petebacondarwin heya, I added the code changes you requested, but unsure of the problems in the prose. Can you give me a bit of direction on that?

**Comment by gkalpak** [comment]
@filipesilva, (as discussed with @petebacondarwin) I will go over this tonight and provide feedack/suggest changes.

**Comment by gkalpak** [comment]
I have rebased on master and pushed a commit with several minor fixes. Other than that LGTM (assuming the code works - I haven't run any code 😇).

**Review by petebacondarwin — APPROVED** [review]

**Comment by gkalpak** [comment]
:tada:

**Comment by filipesilva** [comment]
Awesomesauce, great work everyone 👍

**Pull request merged**